debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/QueueSegment.java

```diff
@@ -109,8 +124,10 @@ public void close() throws IOException {
         writer.close();
     }
 
-    public void sync() throws SyncFailedException {
+    public void sync() throws IOException{
```

Comment:
@amit-yb basically, every time fsync, we write the committed size to metadatadb. 
now fsync happens in two situations:
1. at the end of a batch
2. at the rollover of a segment. 

I remember we discussed the option of only allowing rollover at the end of the batch. Haven't made that change yet because the value-add there is  only of reducing the possibility of a more-than-once delivery (which we thought we will tackle later). LMK if you still want me to do this right now.

Reply:
Will it be too much extra work? If yes, let's postpone it and file a ticket.